setup/main.py

```diff
@@ -198,34 +198,76 @@ def __init_panel(self):
-            current_theme_number = theme_name_list.index(current_theme) + 1
-        except ValueError:
-            current_theme_number = 0
-
+        self.__combobox_custom_theme = self.__builder.get_object(
+                "combobox_custom_theme")
         self.__checkbutton_custom_theme = self.__builder.get_object(
                 "checkbutton_custom_theme")
+
+        def update_combobox_custom_theme(settings, key):
+            theme_name_list = self.__init_available_gtk_themes()
+            self.__model_custom_theme.clear()
+            for name in theme_name_list:
+                self.__model_custom_theme.append([name])
+            current_theme = self.__settings_panel.get_string(key)
+            try:
+                current_theme_number = theme_name_list.index(current_theme)
+            except ValueError:
+                self.__settings_panel.reset(key)
+                current_theme_number = theme_name_list.index(self.__settings_panel.get_string(key))
+            self.__combobox_custom_theme.set_active(current_theme_number)
+
+        update_combobox_custom_theme(None, 'custom-theme')
         self.__settings_panel.bind('use-custom-theme',
                                    self.__checkbutton_custom_theme,
                                    'active',
                                    Gio.SettingsBindFlags.DEFAULT)
-
-        self.__combobox_custom_theme = self.__builder.get_object(
-                "combobox_custom_theme")
-        self.__combobox_custom_theme.set_active(current_theme_number)
+        self.__settings_panel.connect('changed::custom-theme',
+                                   update_combobox_custom_theme)
         self.__settings_panel.bind('use-custom-theme',
                                    self.__combobox_custom_theme,
                                    'sensitive',
                                    Gio.SettingsBindFlags.DEFAULT)
         self.__combobox_custom_theme.connect("changed",
                                    self.__on_combobox_custom_theme_changed)
 
+
+        # custom icon
+        self.__model_custom_icon = self.__builder.get_object(
+                "model_custom_icon")
+        self.__combobox_custom_icon = self.__builder.get_object(
+                "combobox_custom_icon")
+        self.__checkbutton_custom_icon = self.__builder.get_object(
+                "checkbutton_custom_icon")
+
+        def update_combobox_custom_icon(settings, key):
+            icon_name_list = self.__init_available_gtk_icons()
+            self.__model_custom_icon.clear()
+            for name in icon_name_list:
+                self.__model_custom_icon.append([name])
+            current_icon = self.__settings_panel.get_string(key)
+            try:
+                current_icon_number = icon_name_list.index(current_icon)
+            except ValueError:
+                self.__settings_panel.reset(key)
+                current_icon_number = icon_name_list.index(self.__settings_panel.get_string(key))
```

Comment:
Please separate the line.

current_icon = self.__settings_panel.get_string(key)
current_icon_number = icon_name_list.index(current_icon)